ratbagd/ratbagd-resolution.c

```diff
@@ -95,6 +96,65 @@ static int ratbagd_resolution_set_active(sd_bus_message *m,
 	return sd_bus_reply_method_return(m, "u", 0);
 }
 
+static int ratbagd_resolution_disabled_signal_cb(sd_bus *bus,
+						struct ratbagd_resolution *resolution)
+{
+	/* FIXME: we should cache is_disabled and only send the signal for
+	 * those resolutions where it changed */
+
+	(void) sd_bus_emit_properties_changed(bus,
+					      resolution->path,
+					      RATBAGD_NAME_ROOT ".Resolution",
+					      "IsDisabled",
+					      NULL);
+
+	return 0;
+}
+
+static int ratbagd_resolution_set_disabled(sd_bus_message *m,
+					 void *userdata,
+					 sd_bus_error *error)
+{
+	struct ratbagd_resolution *resolution = userdata;
+	int r;
+
+	r = ratbag_resolution_set_disabled(resolution->lib_resolution);
+	if (r < 0) {
+		sd_bus *bus = sd_bus_message_get_bus(m);
+		r = ratbagd_device_resync(resolution->device, bus);
+		if (r < 0)
+			return r;
+	}
+
+	ratbagd_for_each_resolution_signal(sd_bus_message_get_bus(m),
+					   resolution->profile,
+					   ratbagd_resolution_disabled_signal_cb);
+
+	return sd_bus_reply_method_return(m, "u", 0);
+}
+
+static int ratbagd_resolution_set_enabled(sd_bus_message *m,
+					 void *userdata,
+					 sd_bus_error *error)
+{
+	struct ratbagd_resolution *resolution = userdata;
+	int r;
```

Comment:
Did it that way because `ratbagd_resolution_set_active` right above this also uses `r` instead of `rc`, so it's the same everywhere in this file at least. I think it's okay for now but maybe one day we replace all `r`s with `rc`s.

Reply:
Yeah yeah, that's fine then.